### PR TITLE
Fix import of relative file paths from playlists

### PR DESCRIPTION
**Pull request by uklotzde**
https://bugs.launchpad.net/mixxx/+bug/1836243

Tests added and updated.

### CODE REVIEW

**Comment by uklotzde** [comment]
No interest in getting reported bugs fixed??

**Review by daschuer — CHANGES_REQUESTED** [review]
Ups, this was fallen off my priority list.

File: src/library/parser.cpp [hunk 1]
```diff
@@ -146,11 +147,24 @@ bool Parser::isUtf8(const char* string) {
...
-TrackFile Parser::playlistEntryToTrackFile(const QString& playlistEntry) {
+std::pair<TrackFile, bool> Parser::playlistEntryToTrackFile(
+        const QString& playlistEntry,
+        const QString& basePath) {
     if (playlistEntry.startsWith("file:")) {
-        return TrackFile::fromUrl(QUrl(playlistEntry));
+        auto trackFile = TrackFile::fromUrl(QUrl(playlistEntry));
+        return std::make_pair(trackFile, trackFile.checkFileExists());
+    }
+    auto filePath = QString(playlistEntry).replace('\\', '/');
```
[the remark this answers is not in the document]
Comment (uklotzde) [reply src/library/parser.cpp]: Out of the scope of this PR. I won't change any file path parsing functions.

[the remark this answers is not in the document]
Comment (daschuer) [reply src/library/parser.cpp]: OK.

File: src/library/parser.cpp [hunk 2]
```diff
@@ -146,11 +147,24 @@ bool Parser::isUtf8(const char* string) {
     return true;
 }
 
-TrackFile Parser::playlistEntryToTrackFile(const QString& playlistEntry) {
+std::pair<TrackFile, bool> Parser::playlistEntryToTrackFile(
```
[the remark this answers is not in the document]
Comment (uklotzde) [reply src/library/parser.cpp]: I found a simpler solution that doesn't intermingle different aspects.

**Comment by Be-ing** [comment]
Does this affect 2.2?

**Comment by uklotzde** [comment]
It affects 2.2. But it would require a backport and I'm not keen on doing this twice for a functionality that I personally don't need. Another reason for aiming to release 2.3.0 of only to prevent exhausting contributors ;)

**Comment by daschuer** [comment]
LGTM, and successfully tested. Thank you.

**Pull request merged**